lib/device_api/android/device/kindle.rb

```diff
@@ -7,17 +7,21 @@ class Kindle < Device
       # work due to Amazons implementation of the Keyguard.
       def unlock
         ADB.keyevent(qualifier, '26') unless screen_on?
-
         return ADB.swipe(qualifier, {x_from: 900, y_from: 500, x_to: 300, y_to: 500}) if version.split('.').first.to_i < 5
+        ADB.swipe(qualifier, swipe_coords)
+      end
 
-        if orientation == :landscape
-          coords = { x_from: 500, y_from: 750, x_to: 500, y_to: 250 }
+      def swipe_coords
+        res = resolution
+        x = res[1].to_i
+        y = res[2].to_i
+        if version.split('.').first.to_i < 5
```

Comment:
This test was already done above in line 10 and means that this code is never reached.
In the old version this was checking the orientation rather than the version.